DQM/src/TrigScintHitDQM.cxx

```diff
@@ -16,128 +16,86 @@ namespace ldmx {
-
-        histograms_->create<TH2F>("min_time_hit_above_thresh:pe", 
-                                  "Photoelectrons in a TrigScint bar", 1500, 0, 1500, 
-                                  "Earliest time of TrigScint hit above threshold (ns)", 1600, -100, 1500);
-	*/
-
-
-
-	// instead do this 
-
-	histograms_->create<TH1F>("id_"+padName_, "Channel ID of hits", 100, 0, 100);
-	histograms_->create<TH1F>("id_noise_"+padName_, "Channel ID of pure noise hits", 100, 0, 100);
-        histograms_->create<TH1F>("total_pe_"+padName_, "Total PE in the pad/event", 2000, 0, 2000);
-        histograms_->create<TH1F>("n_hits_"+padName_, "TrigScint hit multiplicity in the pad/event", 100, 0, 100);
-        histograms_->create<TH1F>("n_hits_noise_"+padName_, "TrigScint noise hit multiplicity in the pad/event", 100, 0, 100);
-        histograms_->create<TH1F>("x_"+padName_, "Hit x position", 1000, -100, 100);
-        histograms_->create<TH1F>("y_"+padName_, "Hit y position", 1000, -100, 100);
-        histograms_->create<TH1F>("z_"+padName_, "Hit z position", 5000, -900, 100);
-
-        histograms_->create<TH1F>("pe_"+padName_, "PE in a TrigScint bar", 1500, 0, 1500);
-        histograms_->create<TH1F>("pe_noise_"+padName_, "Pure noise hit PE in a TrigScint bar", 1500, 0, 1500);
-        histograms_->create<TH1F>("hit_time_"+padName_, "TrigScint hit time (ns)", 1600, -100, 1500);
-
-        histograms_->create<TH2F>("max_pe:time_"+padName_,
-                                  "Max Photoelectrons in a TrigScint bar", 1500, 0, 1500,
-                                  "TrigScint max PE hit time (ns)", 1500, 0, 1500);
-
-        histograms_->create<TH2F>("min_time_hit_above_thresh:pe_"+padName_,
-                                  "Photoelectrons in a TrigScint bar", 1500, 0, 1500,
-                                  "Earliest time of TrigScint hit above threshold (ns)", 1600, -100, 1500);
-
-
-
+        std::cout << "Process starts!" << std::endl;
+
+        getHistoDirectory();
+    
+        histograms_.create("id", "Channel ID of sim hit", 100, 0, 100);
+        histograms_.create("total_pe", "Total pe deposition in the pad/event", 2000, 0, 2000);
+        histograms_.create("n_hits", "TrigScint hit multiplicity in the pad/event", 100, 0, 100);
+        histograms_.create("x", "Hit x position", 1000, -100, 100);
+        histograms_.create("y", "Hit y position", 1000, -100, 100);
+        histograms_.create("z", "Hit z position", 1000, -900, 100);
+        
+        histograms_.create("pe", "Pe deposition in a TrigScint bar", 1500, 0, 1500);
+        histograms_.create("hit_time", "TrigScint hit time (ns)", 1600, -100, 1500);
+        
+        histograms_.create("id_noise", "Channel ID of noise hit", 100, 0, 100);
+        histograms_.create("pe_noise", "Pe deposition in a TrigScint bar noise hit", 1500, 0, 1500);
+        histograms_.create("n_hits_noise", "TrigScint noise hit multiplicity in the pad/event", 100, 0, 100);
 
```

Comment:
So, can I take it that you have achieved what I was aiming for, namely, if I in my config specify 
`tag = ldmxcfg.Analyzer("Tag", "ldmx::TrigScintHitDQM")`
and
`up = ldmxcfg.Analyzer("Up", "ldmx::TrigScintHitDQM")`,
the histograms will end up getting names starting with Tag and Up, respectively, keeping them unique?

I guess what is unclear to me is if you meant that it's the name of the instance of the processor, or the name of the processor class, that gets prepended to all histogram names.

Reply:
Yes, the `HistogramHelper` class prepends the instance name of the processor (so Tag or Up) to the histograms.